### PR TITLE
JAMES-2702 TimeConverter should not convert duration to seconds but use java.tim…

### PR DESCRIPTION
**Pull request by mbaechler**
…e types

### CODE REVIEW

File: examples/custom-mailets/src/main/java/org/apache/james/examples/custom/mailets/IsDelayedForMoreThan.java [hunk 1]
```diff
@@ -60,7 +61,7 @@ public IsDelayedForMoreThan() {
     @Override
     public void init() {
         String condition = getCondition();
-        maxDelay = Duration.ofMillis(TimeConverter.getMilliSeconds(condition, DEFAULT_UNIT));
+        maxDelay = TimeConverter.parseDuration(condition, DEFAULT_UNIT);
```
Comment (chibenwa) [inline examples/custom-mailets/src/main/java/org/apache/james/examples/custom/mailets/IsDelayedForMoreThan.java]: IMO TimeConverter -> DurationParser cf https://github.com/linagora/james-project/pull/2258/commits/b1cfc32f434169efb83db0b9e69fd946955c5327

File: server/container/util/src/test/java/org/apache/james/util/TimeConverterTest.java [hunk 2]
```diff
@@ -21,456 +21,160 @@
...
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 
 import org.junit.jupiter.api.Test;
 
-public class TimeConverterTest {
+class TimeConverterTest {
 
     @Test
     void getMilliSecondsShouldConvertValueWhenNoUnitAmountAsString() {
```
Comment (chibenwa) [inline server/container/util/src/test/java/org/apache/james/util/TimeConverterTest.java]: See my awesome parametrized tests https://github.com/linagora/james-project/pull/2258/commits/7f94c03624ccb1b0970746dc15f6a7014fb1748b

File: server/container/util/src/test/java/org/apache/james/util/TimeConverterTest.java [hunk 3]
```diff
@@ -21,456 +21,160 @@
...
-    @Test
-    void getMilliSecondsShouldConvertValueWhenMUnit() {
-        //Given
-        long expected = TimeUnit.MINUTES.toMillis(2);
-        //When
-        long actual = TimeConverter.getMilliSeconds(2, "m");
-        //Then
-        assertThat(actual).isEqualTo(expected);
+        Duration actual = TimeConverter.parseDuration("2 secs");
+        assertThat(actual).isEqualTo(Duration.ofSeconds(2));
```
Comment (chibenwa) [inline server/container/util/src/test/java/org/apache/james/util/TimeConverterTest.java]: Why can't we do 

```
assertThat(TimeConverter.parseDuration("2 secs"))
   .isEqualTo(Duration.ofSeconds(2));
```

File: server/container/util/src/test/java/org/apache/james/util/TimeConverterTest.java [hunk 4]
```diff
@@ -21,456 +21,160 @@
...
-        //When
-        long actual = TimeConverter.getMilliSeconds(2, "minutes");
-        //Then
-        assertThat(actual).isEqualTo(expected);
+        Duration actual = TimeConverter.parseDuration("2 Minute");
+        assertThat(actual).isEqualTo(Duration.ofMinutes(2));
     }
 
     @Test
     void getMilliSecondsShouldConvertValueWhenMinutesAmountAsString() {
```
Comment (chibenwa) [inline server/container/util/src/test/java/org/apache/james/util/TimeConverterTest.java]: s/ getMilliSecondsShould.../parseDurationShould.../ everywhere!

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesOutboxFlagUpdateTest.java [hunk 5]
```diff
@@ -84,7 +85,7 @@ public String getName() {
                 }
 
                 @Override
-                public void enQueue(Mail mail, long delay, TimeUnit unit) {
+                public void enQueue(Mail mail, Duration delay) {
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesOutboxFlagUpdateTest.java]: Nice ^^

File: server/queue/queue-api/src/main/java/org/apache/james/queue/api/MailQueue.java [hunk 6]
```diff
@@ -79,7 +92,9 @@
      * @param unit
      * @throws MailQueueException
      */
-    void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueException;
+    default void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueException {
```
Comment (chibenwa) [inline server/queue/queue-api/src/main/java/org/apache/james/queue/api/MailQueue.java]: Shouldn't this be deprecated?

**Comment by chibenwa** [comment]
@mbaechler I would agree to merge your work fast, rebase my PR (which will solve the comments I did here)

Ok?

**Comment by mbaechler** [comment]
> @mbaechler I would agree to merge your work fast, rebase my PR (which will solve the comments I did here)

Fine

**Comment by Arsnael** [comment]
```
[115800a7bb2d6b185cc61e223c5797941832a090] [ERROR] enqueueWithVeryLongDelayShouldDelayMail  Time elapsed: 0.011 s  <<< ERROR!
[115800a7bb2d6b185cc61e223c5797941832a090] java.time.DateTimeException: Invalid value for Year (valid values -999999999 - 999999999): 292277026646
[115800a7bb2d6b185cc61e223c5797941832a090] 	at org.apache.james.queue.api.DelayedMailQueueContract.enqueueWithVeryLongDelayShouldDelayMail(DelayedMailQueueContract.java:82)
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**